File: apps/admin/src/utils/project-stage.ts
import type { ProjectStage } from "@/api/project";

export const projectStageLabels: Record<ProjectStage, string> = {
  INITIATED: "立项",
  BIDDING: "招投标",
  ENTRY: "进场",
  CONSTRUCTION: "施工",
  COMMISSIONING: "调试",
  INITIAL_ACCEPTANCE: "初验",
  FINAL_ACCEPTANCE: "终验",
  WARRANTY: "质保",
  CLOSED: "结项",
};

export function projectStageLabel(value?: string) {
  if (!value) return "-";
  return projectStageLabels[value as ProjectStage] || value;
}

export function projectStageColor(value?: string) {
  return (
    {
      INITIATED: "default",
      BIDDING: "cyan",
      ENTRY: "blue",
      CONSTRUCTION: "geekblue",
      COMMISSIONING: "purple",
      INITIAL_ACCEPTANCE: "orange",
      FINAL_ACCEPTANCE: "gold",
      WARRANTY: "green",
      CLOSED: "default",
    } as Record<string, string>
  )[value || ""];
}

File: services/api/src/main/java/com/company/ops/api/modules/project/domain/ProjectStage.java
package com.company.ops.api.modules.project.domain;

/**
 * 项目交付阶段，自带顺序与基准进度，避免在服务层硬编码 switch。
 */
public enum ProjectStage {
  INITIATED(0, 0),
  BIDDING(1, 0),
  ENTRY(2, 0),
  CONSTRUCTION(3, 20),
  COMMISSIONING(4, 45),
  INITIAL_ACCEPTANCE(5, 65),
  FINAL_ACCEPTANCE(6, 85),
  WARRANTY(7, 100),
  CLOSED(8, 100);

  private final int order;
  private final int progress;

  ProjectStage(int order, int progress) {
    this.order = order;
    this.progress = progress;
  }

  public int order() {
    return order;
  }

  public int progress() {
    return progress;
  }

  public ProjectStage next() {
    for (ProjectStage stage : values()) {
      if (stage.order == order + 1) return stage;
    }
    return null;
  }

  public ProjectStage previous() {
    for (ProjectStage stage : values()) {
      if (stage.order == order - 1) return stage;
    }
    return null;
  }
}
